src/main/scala/org/apache/spark/metrics/source/custom/AccumulatorMetricsSource.scala

```diff
@@ -0,0 +1,54 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package org.apache.spark.metrics.source.custom
+
+import java.{lang => jl}
+
+import com.codahale.metrics.{Gauge, Meter, MetricRegistry}
+import org.apache.spark.metrics.source.Source
+import org.apache.spark.util.AccumulatorV2
+
+import scala.collection.mutable
+
+/**
+  * Accumulator-based metric source
+  * <p><p>
+  * Usage:
+  * <pre>{@code
+  * val metricsSource = new AccumulatorMetricsSource("prefix")
+  * val metric = spark.sparkContext.longAccumulator("metric-name")
+  * metricsSource.registerAccumulator(metric, metric.name.getOrElse("UNNAMED_ACCUMULATOR"))
+  * SparkEnv.get.metricsSystem.registerSource(metricsSource)
+  * }</pre>
+  * @param prefix for metrics collected by this source
+  */
+class AccumulatorMetricsSource(prefix: String) extends Source {
+  override def sourceName: String = prefix
+  override def metricRegistry: MetricRegistry = registry
+
+  private val registry = new MetricRegistry
+  private val gauges = mutable.Map[String, Long]()
+  private val meters = mutable.Map[String, Meter]()
+
+  def registerAccumulator(acc: AccumulatorV2[jl.Long, jl.Long], name: String) {
+    gauges.put(name, 0L)
```

Comment:
I don't think we need a gauge at all. A Dropwizard Meter emits five separate metrics:

- m1, m5, and m15 rates
- mean rate
- [count](https://github.com/dropwizard/metrics/blob/3.2-maintenance/metrics-core/src/main/java/com/codahale/metrics/Meter.java#L79-L82)

I believe the gauge here is redundant with the meter's `count`.